File: java/src/com/company/ListNode.java
package com.company;


public class ListNode {

    public int val;
    public ListNode next;


    public ListNode() {}
    public ListNode(int val) {
        this.val = val;
    }

    public ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }

    public static ListNode buildLinkedList(int[] values) {
        if (values.length == 0) return new ListNode(0);
        var root = new ListNode(values[0]);
        ListNode cur = root;
        int i = 1;
        while (i < values.length) {
            var nn = new ListNode(values[i]);
            cur.next = nn;
            cur = nn;
            i++;
        }
        return root;
    }

    @Override
    public String toString() {
        return "Node(" + val + ")";
    }
}

File: kotlin/src/main/kotlin/ListNode.kt
class ListNode(
    var `val`: Int,
    var next: ListNode? = null
) {

    override fun toString(): String {
        return "Node($`val`)"
    }

    companion object {
        fun buildLinkedList(values: IntArray): ListNode {
            if (values.isEmpty()) return ListNode(0)
            val root = ListNode(values[0])
            var cur = root
            var i = 1
            while (i < values.size) {
                val nn = ListNode(values[i])
                cur.next = nn
                cur = nn
                i++
            }
            return root
        }
    }
}